src/logging.jl

```diff
@@ -0,0 +1,64 @@
+# Logging used by StructuralIdentifiability.jl
+# Consists of two parts:
+#   - si_logger (using Logging package) for logs
+#   - runtime_logger dictionary for recording runtime info
+#
+#   Defines functions to refresh both + specific logging functions
+#   info_si, debug_si, warn_si, and error_si calling @info, @debug, @warn, and @error, respectively
+
+# not the exhaustive list, just the ones to be initialized
+const _runtime_rubrics = (
+    :id_calls_to_gb,
+    :id_groebner_time,
+    :id_inclusion_check_mod_p,
+    :id_npoints_degree,
+    :id_npoints_interpolation,
+    :id_beautifulization,
+)
+
+const _runtime_logger = Dict(
+    :id_calls_to_gb => 0,
+    :id_groebner_time => 0.0,
+    :id_inclusion_check_mod_p => 0,
+    :id_npoints_degree => 0,
+    :id_npoints_interpolation => 0,
+    :id_beautifulization => 0,
+)
+
+const _si_logger =
+    Ref{Logging.ConsoleLogger}(Logging.ConsoleLogger(Logging.Info, show_limited = false))
+
+function restart_logging(; loglevel = Logging.Info)
+    _si_logger[] = Logging.ConsoleLogger(loglevel, show_limited = false)
+    for r in _runtime_rubrics
+        _runtime_logger[r] = 0
+    end
+end
+
+function info_si(s)
+    with_logger(_si_logger[]) do
+        @info s
+        flush(stdout)
```

Comment:
Default logging stream is probably `stderr`:
https://github.com/JuliaLang/julia/blob/16e61e25c7e15280b1e3c11b549b410e324175c4/stdlib/Logging/src/ConsoleLogger.jl#L121

Reply:
Good point! In fact, ideally I would pass an argument to the logger to force flush after each logging event. Based on [this discussion](https://discourse.julialang.org/t/flushing-output-for-logging-immediately-without-buffering/26800) this may not be possible for the standard logger but possible for others

Reply:
Hm. But it should be fine to flush after each log, as it is done here?

Reply:
Now that I benchmark it, it is probably not fine to flush unconditionally for each `@debug` statement, since it is quite slow:
```julia
julia> @benchmark flush($stderr)
BenchmarkTools.Trial: 10000 samples with 7 evaluations.
 Range (min … max):  4.257 μs … 23.671 μs  ┊ GC (min … max): 0.00% … 0.00%
 Time  (median):     4.329 μs              ┊ GC (median):    0.00%
 Time  (mean ± σ):   4.629 μs ±  1.092 μs  ┊ GC (mean ± σ):  0.00% ± 0.00%

  █▅▂▂▂▁    ▂▂                                               ▁
  ███████▇▆▆███▇▆▆▆▆▅▄▄▅▄▅▅▆▅▆▆▆▆▅▆▅▅▅▅▅▅▅▅▅▄▆▃▄▅▄▅▅▅▅▅▄▅▄▅▄ █
  4.26 μs      Histogram: log(frequency) by time     9.69 μs <

 Memory estimate: 48 bytes, allocs estimate: 2.
```

Reply:
Yeah, I was talking about the idea with wrapping the whole thing into `with_logger` and using the standard macro. Then indeed, manual flushes seem to be a better option, although I do not have a systematic way to put them really.